tempesta_fw/http_sess.c

```diff
@@ -207,6 +206,28 @@ tfw_http_sticky_build_redirect(TfwHttpReq *req, StickyVal *sv, RedirMarkVal *mv)
 	if (mv)
 		tfw_http_redir_mark_prepare(mv, m_buf, sizeof(m_buf), m_chunks,
 					    sizeof(m_chunks), &rmark);
+
+	sticky = req->vhost->cookie;
+	if (sticky->js_challenge && jsch_allow) {
+		if (mv) {
+			body = tfw_strdup_desc(req->pool, &sticky->js_challenge->body);
```

Comment:
It seems here and at the below are extra memory allocations, which can be replaced with automatic variables